quesma/clickhouse/schema_loader.go

```diff
@@ -92,31 +96,55 @@ func (sl *tableDiscovery) ReloadTableDefinitions() {
 		sl.tableDefinitions.Store(NewTableMap())
 		return
 	} else {
-		for table, columns := range tables {
-			if indexConfig, found := sl.cfg.IndexConfig[table]; found {
-				if indexConfig.Enabled {
-					for colName := range columns {
-						if _, exists := indexConfig.Aliases[colName]; exists {
-							logger.Error().Msgf("column [%s] clashes with an existing alias, table [%s]", colName, table)
+		if sl.TableAutodiscoveryEnabled() {
+			var autoDiscoResults strings.Builder
```

Comment:
Just for readability maybe we could extract separated functions for both branches

Reply:
https://github.com/QuesmaOrg/quesma/pull/497/commits/d7af1dc3cd7e0cbfe3b501866a32681f24ddcde3